File: server/StoreDesk.API/Models/Transaction.cs
namespace StoreDesk.API.Models;

public class Transaction
{
    public int Id { get; set; }
    public int ItemId { get; set; }
    public Item? Item { get; set; }
    public string BorrowerName { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public DateTime IssuedAt { get; set; }
    public DateTime? ReturnedAt { get; set; }
    public bool IsReturned { get; set; }
}

File: client/src/types/transaction.ts
export interface Transaction {
  id: number;
  borrowerName: string;
  borrowerEmail: string;
  itemId: number;
  itemName?: string;
  quantity: number;
  borrowedAt: string;
  returnedAt?: string;
  isReturned: boolean;
}

export interface CreateTransactionPayload {
  borrowerName: string;
  borrowerEmail: string;
  itemId: number;
  quantity: number;
}
